### PR TITLE
Add __ tag in bold parser and test

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: include/maddy/strongparser.h [hunk 1]
```diff
@@ -41,10 +41,13 @@ class StrongParser : public LineParser
   void
   Parse(std::string& line) override
   {
-    static std::regex re("(?!.*`.*|.*<code>.*)\\*\\*(?!.*`.*|.*<\\/code>.*)([^\\*\\*]*)\\*\\*(?!.*`.*|.*<\\/code>.*)");
+    static std::vector<std::regex> res;
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\*\\*(?!.*`.*|.*<\\/code>.*)([^\\*\\*]*)\\*\\*(?!.*`.*|.*<\\/code>.*)"});
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\_\\_(?!.*`.*|.*<\\/code>.*)([^\\_\\_]*)\\_\\_(?!.*`.*|.*<\\/code>.*)"});
```
Comment (progsource) [inline include/maddy/strongparser.h]: Whenever Parse is called, res will get 2 additional entries, so instead use an initializer list here.

File: include/maddy/strongparser.h [hunk 2]
```diff
@@ -41,10 +41,13 @@ class StrongParser : public LineParser
...
     static std::string replacement = "<strong>$1</strong>";
-
-    line = std::regex_replace(line, re, replacement);
+    for(const auto& re : res) {
```
Comment (progsource) [inline include/maddy/strongparser.h]:
```suggestion
    for (const auto& re : res)
    {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 3]
```diff
@@ -12,33 +12,78 @@
...
 {
-  std::string text = "some text **bla** text testing **it** out";
-  std::string expected = "some text <strong>bla</strong> text testing <strong>it</strong> out";
-  auto strongParser = std::make_shared<maddy::StrongParser>();
+  struct testIt {
+    std::string text;
+    std::string expected;
+  };
+
+  std::vector<testIt> tests{
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  std::vector<testIt> tests
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 4]
```diff
@@ -12,33 +12,78 @@
...
+      "some text __bla__ text testing __it__ out",
+      "some text <strong>bla</strong> text testing <strong>it</strong> out"
+    },
+  };
 
-  strongParser->Parse(text);
+  auto strongParser = std::make_shared<maddy::StrongParser>();
 
-  ASSERT_EQ(expected, text);
+  for(auto& test : tests) {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  for (auto& test : tests)
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 5]
```diff
@@ -12,33 +12,78 @@
 
 TEST(MADDY_STRONGPARSER, ItReplacesMarkdownWithStrongHTML)
 {
-  std::string text = "some text **bla** text testing **it** out";
-  std::string expected = "some text <strong>bla</strong> text testing <strong>it</strong> out";
-  auto strongParser = std::make_shared<maddy::StrongParser>();
+  struct testIt {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  struct testIt
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 6]
```diff
@@ -12,33 +12,78 @@
...
+    ASSERT_EQ(test.expected, test.text);
+  }
 }
 
 TEST(MADDY_STRONGPARSER, ItReplacesEmphasizedMarkdownNotWithStrongHTML)
 {
-  std::string text = "some text *bla* text testing **it** out";
-  std::string expected = "some text *bla* text testing <strong>it</strong> out";
-  auto strongParser = std::make_shared<maddy::StrongParser>();
+  struct testIt {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  struct testIt
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 7]
```diff
@@ -12,33 +12,78 @@
...
+    {
+      "some text _bla_ text testing __it__ out",
+      "some text _bla_ text testing <strong>it</strong> out"
+    },
+  };
 
-  ASSERT_EQ(expected, text);
+  auto strongParser = std::make_shared<maddy::StrongParser>();
+
+  for(auto& test : tests) {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  for (auto& test : tests)
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 8]
```diff
@@ -12,33 +12,78 @@
...
+    ASSERT_EQ(test.expected, test.text);
+  }
 }
 
 TEST(MADDY_STRONGPARSER, ItDoesNotParseInsideInlineCode)
 {
-  std::string text = "some text **bla** `/**text**/` testing `**it**` out";
-  std::string expected = "some text **bla** `/**text**/` testing `**it**` out";
-  auto strongParser = std::make_shared<maddy::StrongParser>();
+  struct testIt {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  struct testIt
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 9]
```diff
@@ -12,33 +12,78 @@
...
+    std::string text;
+    std::string expected;
+  };
 
-  strongParser->Parse(text);
+  std::vector<testIt> tests{
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  std::vector<testIt> tests
  {
```

File: tests/maddy/test_maddy_strongparser.cpp [hunk 10]
```diff
@@ -12,33 +12,78 @@
...
+    {
+      "some text _bla_ text testing __it__ out",
+      "some text _bla_ text testing <strong>it</strong> out"
+    },
+  };
+
+  auto strongParser = std::make_shared<maddy::StrongParser>();
 
-  ASSERT_EQ(expected, text);
+  for(auto& test : tests) {
```
Comment (progsource) [inline tests/maddy/test_maddy_strongparser.cpp]:
```suggestion
  for (auto& test : tests)
  {
```

**Comment by patrickelectric** [comment]
Hi @progsource, thank you for the review, much appreciate !
Is there a astyle script to helps with the style problems ?
PS: I sent a mail via info [at] progsource, but gmail failed to delivery.

**Comment by patrickelectric** [comment]
k np. Comment block updated.

**Comment by progsource** [comment]
Thank you very much.

**Pull request merged**